plugins/modules/pacemaker_stonith.py

```diff
@@ -0,0 +1,228 @@
+            )),
+            agent_validation=dict(type='bool', default=False),
+            wait=dict(type='int', default=300),
+        ),
+        required_if=[('state', 'present', ['stonith_type', 'stonith_options'])],
+        supports_check_mode=True
+    )
+
+    use_old_vardict = False
+    default_state = "present"
+
+    def __init_module__(self):
+        self.runner = pacemaker_runner(self.module, cli_action='stonith')
+        self.vars.set('previous_value', self._get())
+        self.vars.set('value', self.vars.previous_value, change=True, diff=True)
+
+    def _process_command_output(self, fail_on_err, ignore_err_msg=""):
+        def process(rc, out, err):
+            if fail_on_err and rc != 0 and err and ignore_err_msg not in err:
+                self.do_raise('pcs failed with error (rc={0}): {1}'.format(rc, err))
+            out = out.rstrip()
+            return None if out == "" else out
+        return process
+
+    def _get(self):
+        with self.runner('state name', output_process=self._process_command_output(False)) as ctx:
+            return ctx.run(state='status')
+
+    def _fmt_stonith_resource(self):
+        return dict([("resource_name", self.vars.stonith_type)])
+
+    # TODO: Pluralize operation_options in separate PR and remove this helper fmt function
+    def _fmt_stonith_operations(self):
+        modified_stonith_operations = []
+        for stonith_operation in self.vars.stonith_operations:
+            modified_stonith_operations.append(dict([("operation_action", stonith_operation.get('operation_action')),
+                                                    ("operation_option", stonith_operation.get('operation_options'))]))
+        return modified_stonith_operations
+
+    def state_absent(self):
+        with self.runner('state name', output_process=self._process_command_output(True, "does not exist"), check_mode_skip=True) as ctx:
+            ctx.run()
+            self.vars.set('value', self._get())
+            self.vars.stdout = ctx.results_out
+            self.vars.stderr = ctx.results_err
+            self.vars.cmd = ctx.cmd
+
+    def state_present(self):
+        with self.runner(
+                'state name resource_type resource_option resource_operation resource_meta resource_argument agent_validation wait',
+                output_process=self._process_command_output(True, "already exists"),
+                check_mode_skip=True) as ctx:
+            ctx.run(resource_type=self._fmt_stonith_resource(),
+                    resource_option=self.vars.stonith_options,
+                    resource_operation=self._fmt_stonith_operations(),
+                    resource_meta=self.vars.stonith_metas,
+                    resource_argument=self.vars.stonith_argument)
+            self.vars.set('value', self._get())
+            self.vars.stdout = ctx.results_out
+            self.vars.stderr = ctx.results_err
+            self.vars.cmd = ctx.cmd
```

Comment:
If you do this in the output_process function, no need to repeat yourself - keep it DRY